pyomo/contrib/edi/tests/test_blackbox.py

```diff
@@ -0,0 +1,958 @@
+#  ___________________________________________________________________________
+#
+#  Pyomo: Python Optimization Modeling Objects
+#  Copyright (c) 2008-2023
+#  National Technology and Engineering Solutions of Sandia, LLC
+#  Under the terms of Contract DE-NA0003525 with National Technology and
+#  Engineering Solutions of Sandia, LLC, the U.S. Government retains certain
+#  rights in this software.
+#
+#  Development of this module was conducted as part of the Institute for
+#  the Design of Advanced Energy Systems (IDAES) with support through the
+#  Simulation-Based Engineering, Crosscutting Research Program within the
+#  U.S. Department of Energy’s Office of Fossil Energy and Carbon Management.
+#
+#  This software is distributed under the 3-clause BSD License.
+#  ___________________________________________________________________________
+
+import pyomo.common.unittest as unittest
+import pyomo.environ as pyo
+from pyomo.common.dependencies import attempt_import
+
+from pyomo.core.base.units_container import pint_available
+
+np, numpy_available = attempt_import('numpy', 'edi requires numpy')
+# scipy, scipy_available = attempt_import(
+#     'scipy', 'inverse_reduced_hessian requires scipy'
+# )
+
+# if not (numpy_available and scipy_available):
+if not numpy_available:
+    raise unittest.SkipTest('edi.formulation tests require numpy')
+
+
+@unittest.skipIf(not pint_available, 'Testing units requires pint')
+class TestEDIBlackBox(unittest.TestCase):
+    def test_edi_blackbox_variable(self):
+        "Tests the black box variable class"
+        from pyomo.environ import units
+        from pyomo.contrib.edi import Formulation
+        from pyomo.contrib.edi.blackBoxFunctionModel import (
+            BlackBoxFunctionModel_Variable,
+            TypeCheckedList,
+            BBList,
+            BlackBoxFunctionModel,
+        )
```

Comment:
Imports should be at the top in most cases

Reply:
I prefer to have unit tests be entirely self contained (ie, I can copy and paste it and it will run).  However, I am willing to adapt if this massively violates a core pyomo philosophy